lib/engine/game/g_18_eus/step/bidbox_auction.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/auctioner'
+
+module Engine
+  module Game
+    module G18EUS
+      module BidboxAuction
+        include Engine::Step::Auctioner
+
+        def actions(entity)
+          actions = []
+          actions << 'bid' << 'pass' if !@game.buyable_bank_owned_companies.empty?
+          actions.concat(super).uniq
+        end
+
+        def setup
+          # This sets the initial value of @bids
+          setup_auction
+          super
+
+          @bid_actions = 0
+          @bids = @round.bids if @round.bids
+          # Set initial value of @bids on the round if there's none.
+          @round.bids = @bids unless @round.bids
+        end
+
+        def can_buy_company?(_player, _company)
+          false # Only companies are privates
+        end
+    
+        def min_bid(company)
+          return unless company
+
+          high_bid = highest_bid(company)
+          (high_bid ? high_bid.price + min_increment : company.min_bid)
+        end
+
+        def max_bid(player, _company)
+          player.cash - committed_cash(player)
+        end
+
+        def committed_cash(player, _show_hidden = false)
+          bids_for_player(player, only_committed_bids: true).sum(&:price)
+        end
+
+        def find_bid(player, company)
+          @bids[company]&.find { |b| b.entity == player }
+        end
+
+        def highest_player_bid?(player, company)
+          return false unless find_bid(player, company)
+
+          current_bid_amount(player, company) >= (highest_bid(company)&.price || 0)
+        end
+
+        protected
+
+        def active_auction
+          company = @auctioning
+          bids = @bids[company]
+          yield company, bids if bids.size > 1
+        end
+
+        def bids_for_player(player, only_committed_bids: false)
+          @bids.values.map do |bids|
+            if only_committed_bids
+              highest_bid = bids.max_by(&:price)
+              highest_bid if highest_bid&.entity == player
+            else
+              bids.find { |bid| bid.entity == player }
+            end
+          end.compact
+        end
+
+        def num_certs_with_bids(entity)
```

Comment:
I don't think this matters for us. Privates don't count as certs, right? Can you check the rules?

Reply:
●	A “certificate” can be understood as one card, which includes one single share of a railroad company, the president’s certificate (which is two shares), one private company, or one bank share.

Privates do count as certs